### DISCUSSION TITLE
How to use Laravel 9 Factory on package Models?

### DISCUSSION THREAD

**Discussion by AleksandrovC** [cmgmyr/laravel-messenger#384] [Q&A] [answered]
Hi and thanks for the all the work on this package!

I have the following question:
Is there any way to use Laravel Factory on Threads, Messages and Participants Models?

I saw in Laravel documentation that i have to add `use HasFactory;` at the begging of the Model - but this means to modify the package Models.

Is this a bad practice - should I do it any other way?

**Comment by cmgmyr** [comment, marked as answer]
Hi and thanks for using the package!

Unfortunately, the package isn't currently set up for Laravel 8+ factories. This would be a great addition but I would want this to be included with some other updates and a major version bump.

In the meantime, a workaround would be to create your own models and extend each one from the package. Remember to update the config file too. In your own models, you can now use the `HasFactory` trait, then go ahead and make your factory classes.

I hope that helps.
